packages-internal/test-utils/src/describeConformance.tsx

```diff
@@ -1034,8 +1034,8 @@ function describeConformance(
     });
     window.matchMedia = () =>
       ({
-        addListener: () => {},
-        removeListener: () => {},
+        addEventListener: () => {},
+        removeEventListener: () => {},
```

Comment:
On second thought, it should mock both here to work with both v5 and v6.

Reply:
Not sure I follow. This PR is supposed to land on `next` (v6). How does keeping `addListener`/`removeListener` help v5?

Reply:
Given the current release strategy for the internal packages, this should probably be a major release of the `@mui/internal-test-utils` and dropping the `addListener` altogether as the linked PR for removing Safari support did. 🤔 
What do others think?
cc @mui/code-infra

Reply:
> This PR is supposed to land on next (v6). How does keeping addListener/removeListener help v5?

We're using the `@mui/internal-test-utils` released from the `next` branch, but we can run our tests using either `@mui/material` v5 or v6. Since `@mui/material@5` still uses `addListener`/`removeListener`, we need both to run the tests against both v5 and v6.
Does this make sense?

Reply:
Got it 👍

Reply:
> Given the current release strategy for the internal packages, this should probably be a major release of the `@mui/internal-test-utils`

We're not following semver with internal packages. The strategy is to be always backwards compatible (*). So I believe supporting both as @cherniavskii does makes most sense to me 👍 


_(*) backwards compatible against our own code that is. As soon as none of our code uses the compatibility mode anymore, it can be dropped, even on a minor._

Reply:
Sounds good 👍🏻
This PR is ready for review then.